src/gt4py/next/ffront/fbuiltins.py

```diff
@@ -133,6 +133,19 @@ def builtin_function(fun: Callable[_P, _R]) -> BuiltInFunction[_R, _P]:
     return BuiltInFunction(fun)
 
 
+class WhereBuiltinFunction(BuiltInFunction[_R, _P]):
+    def dispatch(self, *args: Any) -> Callable[_P, _R]:
+        arg_types = tuple(type(arg) for arg in args)
+        if arg_types[1] == tuple and arg_types[2] == tuple:
+            # TODO(havogt): temporary workaround for `where(cond, tuple(...), tuple(...))`
+            return self.function
+        return super().dispatch(*args)
```

Comment:
```suggestion
class WhereBuiltinFunction(BuiltInFunction[_R, _P]):
    def __call__(self, *args: Any) -> Callable[_P, _R]:
      if isinstance(true_field, tuple) or isinstance(false_field, tuple):
          if not (isinstance(true_field, tuple) and isinstance(false_field, tuple)):
              raise ValueError(
                  f"Either both or none can be tuple in {true_field=} and {false_field=}."
              )
          if len(true_field) != len(false_field):
              raise ValueError("Tuple of different size not allowed.")
          return tuple(where(mask, t, f) for t, f in zip(true_field, false_field))
      return super()(*args)
```
Why not just have all the implementation here?